### ISSUE TITLE
Support for iPega 9083s

### ISSUE THREAD

**Issue by GabrielMundo** [ndeadly/MissionControl#98]
Hello, as the title says, could you please add support for iPega 9083s?
I've tried all the 3 sync modes, but it just doesn't connect.
btdb only shows my main joycons, nothing else.
I've also tried every "experimental" version of other iPega controller that you have posted in another requests (9087, 9076, etc.), but it's the same, doesn't even connect.
Thanks

**Comment by ndeadly** [comment]
If it's not showing up in btdb then it's not able to complete the pairing process. Try and pair it while running this build to create a log. The log is stored at `sdmc://bluetooth-mitm.log` and is overwritten every time you reboot the console.

Just to be clear, you're trying to pair the controller at the `Change Grip/Order` screen, right?

[MissionControl-0.2.1-debug-logging.zip](https://github.com/ndeadly/MissionControl/files/5263252/MissionControl-0.2.1-debug-logging.zip)

**Comment by GabrielMundo** [comment]
Yes, of course, at the Change grip/order screen.
Here is the log

[bluetooth-mitm.log](https://github.com/ndeadly/MissionControl/files/5263614/bluetooth-mitm.log)

**Comment by ndeadly** [comment]
The switch doesn't seem to have seen any devices at all. Was this log taken trying every pairing mode? Can you verify whether a phone or PC is able to see the controller?

**Comment by ndeadly** [comment]
Which button combo are you using to enter pair mode? There seems to be conflicting information on the [ipega website](http://ipega.hk/gamehandle/53-180.html). The manual says to set the switch and hold HOME for 3 seconds, but the [other document ](http://ipega.hk/uploads/201906/5cff1bfb01208.pdf) suggests you should be pressing A+HOME (which is what most other ipega users are using I think)

**Comment by GabrielMundo** [comment]
Tried all modes (the controller comes with a physical switch to change modes, and also LED indicating each mode), also tried home + A with each mode, but without success. By the way, I waited 30-40 secs on each try.
The controller works fine on my android phone.

Here is the log:
[bluetooth-mitm.log](https://github.com/ndeadly/MissionControl/files/5263884/bluetooth-mitm.log)

I thought that maybe switch's bluetooth was not working, but these Eastvita controllers paired successfully (they do pair without the need of MissionControll)
![6jXQ7hI](https://user-images.githubusercontent.com/53465027/93934339-9de71200-fce8-11ea-80b3-590136d861f3.png)
Here is the log after pairing eastvita controller:
[bluetooth-mitm.log](https://github.com/ndeadly/MissionControl/files/5263880/bluetooth-mitm.log)

**Comment by GabrielMundo** [comment]
The iPega also works on PC, and it's detected as a controller
![y2X156S](https://user-images.githubusercontent.com/53465027/93935806-cf60dd00-fcea-11ea-83e4-c7e30be5781d.png)

**Comment by ndeadly** [comment]
Yeah it doesn't seem to be getting picked up at all. Even if it couldn't pair properly I would at least expect to see it show up in the logs as discovered. Only other thing I can think of is the controller might be using Bluetooth LE (which I don't currently support). I'm not sure how to verify that though. It would be the first of any ipega controller I've encountered so far to use it if that were the case.

**Comment by GabrielMundo** [comment]
You are right.
![9x5hXaA](https://user-images.githubusercontent.com/53465027/93938165-772bda00-fcee-11ea-8862-6258fe00cbf2.png)
Well, I hope someday you add support to BLE.
Anyway, thank you

**Comment by ndeadly** [comment]
Ah that's a bummer, but good to know that there are some ipega controllers that won't work. I do plan to support BLE eventually, it's more a matter of not having a clue how it works right now. It could require a second reverse engineering effort.

**Comment by afonsocappelo** [comment]
Good afternoon, I'm not sure how to talk to you and I almost gave up ahahah, I'm out of joycon and my salvation would be to use the ipega control, but it doesn't have support for my 9077, could you add? I'm using google translator !

**Comment by ndeadly** [comment]
@afonsocappelo have you tried to connect the controller already? Does it show up as connected? Many ipega controllers will work without any changes required, even if not listed under supported controllers (I've only listed the ones that users have confirmed working). 

If you are still having troubles, please open a [new issue](https://github.com/ndeadly/MissionControl/issues/new) for this controller. It may use a different hardware ID that I haven't whitelisted, or it may be another BLE controller like the one in this issue. If you are able to get it to connect to the console, please use [btdb](https://github.com/ndeadly/MissionControl/releases/download/v0.3.0/btdb.nro) to list the paired devices and attach a screenshot to your issue.
